test/secrets_test.rb

```diff
@@ -182,36 +194,52 @@ def response_body(body)
       stub_request(:get, url).to_raise(Vault::HTTPClientError.new('http://foo.com', stub(code: 403)))
       url2 = url.sub!('very/hidden', 'very/secret') || raise
       stub_request(:get, url2).to_raise(Vault::HTTPClientError.new('http://foo.com', stub(code: 403)))
-      e = assert_raises(RuntimeError) { process }
+      e = assert_raises(RuntimeError) { process_secrets }
       e.message.must_include("Error reading key this/is/very/hidden")
       e.message.must_include("Error reading key this/is/very/secret")
     end
 
     describe 'CONSUL_URL' do
       it 'creates a CONSUL_URL secret' do
-        process
+        process_secrets
         File.read("CONSUL_URL").must_equal("http://#{SecretsClient::LINK_LOCAL_IP}:8500")
       end
 
       it 'can be overwritten by the user' do
         File.write('annotations', "secret/CONSUL_URL=\"this/is/very/hidden\"")
-        process
+        process_secrets
         File.read('CONSUL_URL').must_equal 'foo'
       end
     end
 
     describe 'HOST_IP' do
       it 'creates a HOST_IP secret' do
-        process
+        process_secrets
         File.read("HOST_IP").must_equal("10.10.10.10")
       end
 
       it "raises when host ip api call fails" do
         stub_request(:get, "https://foo.bar/api/v1/namespaces/default/pods").
           to_return(status: 500)
-        e = assert_raises(RuntimeError) { process }
+        e = assert_raises(RuntimeError) { process_secrets }
         e.message.must_include("Could not GET https://foo.bar/api/v1/namespaces/default/pod")
       end
     end
   end
+
+  describe "#process_pki_certs" do
+    let(:certificate) { "-----BEGIN CERTIFICATE-----\nimma cert\n-----END CERTIFICATE-----" }
+    let(:reply) { {data: {certificate: certificate, serial: 'test'}}.to_json }
+    let(:url) { +'https://foo.bar:8200/v1/pki/issue/example-com' }
+
+    before do
+      stub_request(:put, url).with { |request| request.body == {common_name: 'example.com'}.to_json }.to_return(response_body(reply))
+    end
+
+    it 'works' do
+      process_pki_certs
+      File.read("pki/example.com/serial").must_equal('test')
+      File.read("pki/example.com/certificate.pem").must_equal(certificate)
+    end
```

Comment:
no edge-cases eh ? 🤔

Reply:
heh, this was just an initial "does it work" test. will be adding edge case tests. thanks for the review!